### ISSUE TITLE
How to avoid OOM while using to_dense_adj on batch data with many edge_index

### ISSUE THREAD

**Issue by PPGGG** [pyg-team/pytorch_geometric#5101]
### 🐛 Describe the bug

How to avoid OOM while using to_dense_adj on batch data with many edge_index

**Comment by rusty1s** [comment]
How big is your adjacency matrix? There is an upper limit on the node sizes when using dense adjacency matrices.

**Comment by PPGGG** [comment]
@rusty1s The adjacency matrix has almost 50,000 nodes.

**Comment by rusty1s** [comment]
How much RAM do you have? This will need around 10GB to store in a dense adjacency matrix. Independent of that, I don't think it is a good idea to store such a large adjacency matrix in dense format. What's your use-case here? :)

**Comment by PPGGG** [comment]
@rusty1s Thanks a lot.  I would like to calculate node degree information inside a batch which uses neighborhood sampling method.

**Comment by rusty1s** [comment]
You can utilize `utils.degree` for this. Note that the node degree during sampling differs from the original node degree though.

**Comment by PPGGG** [comment]
@rusty1s Yeah i have tried `utils.degree`. But it also OOM when deal with batch data for its huge edge_index number.

**Comment by rusty1s** [comment]
That might be an issue with how you encode your `edge_index`. Note that values within it should only range from `0` to `num_nodes - 1`. It cannot hold arbitrary IDs.

**Comment by PPGGG** [comment]
In CPU that's okay, but in GPU it will OOM.

**Comment by rusty1s** [comment]
Ok, but what is `edge_index.max()` and `edge_index.size()`? Is your edge index tensor correctly formatted?

**Comment by PPGGG** [comment]
@rusty1s Yes. I just use Flickr dataset.

**Comment by rusty1s** [comment]
Do you have an example to reproduce? Happy to look into this.
